pep-0681.rst

```diff
@@ -0,0 +1,709 @@
+dataClasses.ts source file [#pyright-impl]_ would be a good starting
+point for understanding the implementation.
+
+The attrs [#attrs-usage]_ and pydantic [#pydantic-usage]_ libraries are
+using the ``__dataclass_transform__`` `alternate form`_.
+
+
+Rejected Ideas
+==============
+
+``auto_attribs`` parameter
+--------------------------
+The attrs library supports an ``auto_attribs`` parameter that
+indicates whether class members decorated with PEP 526 variable
+annotations but with no assignment should be treated as data fields.
+We considered supporting ``auto_attribs`` and a corresponding
+``auto_attribs_default`` parameter. We decided against this because it
+is specific to attrs and appears to be a legacy behavior. Instead of
+supporting this in the new standard, we recommend that the maintainers
+of attrs move away from the legacy semantics and adopt
+``auto_attribs`` behaviors by default.
+
+Django does not support declaring fields using type annotations only,
+so users of this mechanism would need to know that they should always
+supply assigned values.
+
+``cmp`` parameter
+-----------------
+The attrs library supports a bool parameter ``cmp`` that is equivalent
+to setting both ``eq`` and ``order`` to True. This is not supported in
+this proposal. Attrs users should use the dataclass-standard parameter
+names.
+
+``kw_only`` field descriptor parameter
+--------------------------------------
+The attrs library supports a ``kw_only`` parameter for individual
+fields. This is not currently supported in this spec, but it could be
+added in the future if there was sufficient demand.
+
+Automatic field name aliasing
+-----------------------------
+The attrs library performs automatic aliasing of field names that
+start with a single underscore. This proposal omits this behavior.
+
+Alternate field ordering algorithms
+-----------------------------------
+The attrs library currently supports two approaches to ordering the
+fields within a class:
+
+* Dataclass order: The same ordering used by dataclasses. This is the
+  default behavior of the older APIs (ex. attr.s).
+* Method Resolution Order (MRO): This is the default behavior of the
+  newer APIs (ex. define, mutable, frozen). Older APIs (ex. attr.s)
+  can opt into this behavior by specifying ``collect_by_mro=True``.
+
+The resulting field orderings can differ in certain diamond-shaped
+multiple inheritance scenarios.
+
+This proposal does not support any field ordering other than that used
+by dataclass.
+
+Fields redeclared in subclasses
+-------------------------------
+The attrs library differs from stdlib dataclasses in how it
+handles inherited fields that are redeclared in subclasses. The
+dataclass specification preserves the original order, but attrs
+defines a new order based on subclasses. Users of attrs who rely on
+this ordering will not see the expected order of parameters in the
+synthesized ``__init__`` method.
+
+Django primary and foreign keys
```

Comment:
It might be helpful to briefly mention what "primary" and "foreign" keys are in this context.

Reply:
I think that's OK to omit, since Django is already introduced as an ORM package in the abstract.

Reply:
Would it be worth linking to some external coverage of these terms? For example, https://en.wikipedia.org/wiki/Primary_key

Reply:
Maybe just link to the Django docs, https://docs.djangoproject.com/en/4.0/topics/db/models/#automatic-primary-key-fields